legend-pure-core/legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/compiler/postprocessing/processor/AssociationProcessor.java

```diff
@@ -111,6 +111,16 @@ public static Iterable<AbstractProperty<?>> process(Association association, Con
 
         validateAssociationPropertiesRawTypes(association, leftRawType, rightRawType);
 
+        if (leftProperty._defaultValue() != null)
+        {
+            throw new PureCompilationException(association.getSourceInformation(), "Association properties should not have default values defined. Error in property: " + leftProperty.getName());
+        }
+
+        if (rightProperty._defaultValue() != null)
+        {
+            throw new PureCompilationException(association.getSourceInformation(), "Association properties should not have default values defined. Error in property: " + rightProperty.getName());
+        }
```

Comment:
I think these checks more properly belong in AssociationValidator than in AssociationProcessor. Also, we should probably use the source information of the property, rather than the association.